sdk/core/azure-core/inc/azure/core/azure_assert.hpp

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Provide assert macros to use as pre-conditions.
+ *
+ * @remark Asserts are turned ON even if `NDEBUG` is defined.
+ *
+ */
+
+#pragma once
+
+#if defined(NDEBUG)
+#define AZURE_SDK_INTERNAL_IMPLEMENTATION_OVERRIDE_NDEBUG
+#undef NDEBUG
+#endif
+
+#include <cassert>
+#include <stdexcept>
+
+#define AZURE_ASSERT(exp) assert(exp)
```

Comment:
Does this still trigger abort() on retail?  We want the default behavior on retail to be abort()

Reply:
>  We want the default behavior on retail to be abort()

Can the user override this to no longer abort on release?

Reply:
if `NDEBUG` macro is defined (and it's defined in CMake Release build), `assert` is a no-op.

Reply:
I feel it's kind of dangerous that a user will not get any error message if he tries to get the value of a null nullable variable in release build.